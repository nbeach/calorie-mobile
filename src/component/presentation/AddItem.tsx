import React from "react"
import {StyleSheet, Text, TextInput, TouchableOpacity, View} from "react-native"
import {border} from "../../util/StyleUtil"

export interface AddItemStateProps {
    readonly name: string
    readonly calories: number
}

export interface AddItemDispatchProps {
    readonly addItem: (name: string, calories: number) => void
}

export const AddItem = ({addItem, name, calories}: AddItemStateProps & AddItemDispatchProps) => {
    return <View style={styles.container}>
        <TextInput style={styles.item} placeholder={"Item"}/>
        <TextInput keyboardType="numeric" style={styles.calories} placeholder={"Cals"}/>

        <TouchableOpacity style={styles.button} onPress={() => addItem(name, calories)}>
            <Text style={styles.buttonText}>+</Text>
        </TouchableOpacity>
    </View>
}


const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        justifyContent: "space-between",
        padding: 10,
        backgroundColor: "#e4ebee",
    },
    item: {
        width: "60%",
        borderColor: "#AAAAAA",
        borderBottomWidth: 1,
        borderStyle: "solid",
        fontSize: 18,
    },
    calories: {
        width: "15%",
        borderColor: "#AAAAAA",
        borderBottomWidth: 1,
        borderStyle: "solid",
        fontSize: 18,
    },
    button: {
        width: "15%",
        fontSize: 18,
        alignItems: "center",
        backgroundColor: "#1b5fff",
        justifyContent: "center",
        ...border(1, "solid", "#EEEEEE"),
    },
    buttonText: {
        color: "#FFFFFF",
        fontSize: 18,
        fontWeight: "bold",
    },
})

