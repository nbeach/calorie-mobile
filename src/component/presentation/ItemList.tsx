import React from "react"
import {StyleSheet, Text, View} from "react-native"
import {ItemState} from "../../store/state/ItemState"
import {border} from "../../util/StyleUtil"

export interface ItemListProps {
    readonly items: ReadonlyArray<ItemState>
}

export const ItemList = (props: ItemListProps) => {
    return <View style={styles.container}>
        {props.items.map(item => <Text key={item.name} style={styles.item}>{item.name}</Text>)}
    </View>
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
    },
    item: {
        padding: 10,
        width: "100%",
        textAlign: "center",
        fontSize: 18,
        ...border(1, "solid", "#AAAAAA"),
    },

})
