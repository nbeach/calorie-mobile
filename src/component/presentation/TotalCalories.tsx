import React from "react"
import {Text, View} from "react-native"

export interface TotalCaloriesStateProps {
    readonly totalCalories: number
}

export const TotalCalories = ({totalCalories}: TotalCaloriesStateProps) =>
    <View>
        <Text>{totalCalories.toString()}</Text>
    </View>
