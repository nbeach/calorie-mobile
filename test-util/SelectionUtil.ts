import {ShallowWrapper} from "enzyme"
import {ReactTestRenderer} from "react-test-renderer";

export const nodes = (testId: string, wrapper: ShallowWrapper) => wrapper.findWhere(node => node.prop("testID") === testId)
export const node = (testId: string, wrapper: ShallowWrapper) => nodes(testId, wrapper).first()

export const instances = (testId: string, tree: ReactTestRenderer) => tree.root.findAll(thing => thing.props.testID === testId)
export const instance = (testId: string, tree: ReactTestRenderer) => tree.root.find(thing => thing.props.testID === testId)
