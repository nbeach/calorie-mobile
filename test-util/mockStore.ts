import {AppState} from "../src/store/state/AppState"

export const mockStore = (state: AppState) => ({
    getState: () => state,
    dispatch: jest.fn(),
    subscribe: jest.fn(),
    replaceReducer: jest.fn(),
})
