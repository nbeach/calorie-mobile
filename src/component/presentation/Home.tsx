import React from 'react';
import {ItemList} from "./ItemList";
import {StyleSheet, View} from 'react-native';
import {AppState} from "../../store/state/AppState";
import {AddItemContainer} from "../container/AddItemContainer";

export interface HomeProps extends AppState{

}

export const Home = (props: HomeProps) => {
    return <View style={styles.container}>
        <AddItemContainer></AddItemContainer>
        <ItemList items={props.items}/>
    </View>
};


const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-evenly',
        alignItems: 'stretch',
    },
});
