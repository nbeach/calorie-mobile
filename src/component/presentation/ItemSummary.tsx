import {ItemState} from "../../store/state/ItemState"
import {View} from "react-native"
import {TotalCalories} from "./TotalCalories"
import {ItemList} from "./ItemList"
import React from "react"

export interface ItemSummaryStateProps {
    readonly totalCalories: number,
    readonly items: ReadonlyArray<ItemState>
}

export interface ItemSummaryDispatchProps {
    readonly removeItem: (id: string) => void
}

export const ItemSummary = ({totalCalories, items, removeItem}: ItemSummaryStateProps & ItemSummaryDispatchProps) =>
    <View>
        <TotalCalories totalCalories={totalCalories}/>
        <ItemList items={items} removeItem={removeItem}/>
    </View>
