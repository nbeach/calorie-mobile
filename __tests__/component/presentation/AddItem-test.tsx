import React from "react"

import renderer from "react-test-renderer"
import {AddItem, AddItemTestIds} from "../../../src/component/presentation/AddItem"
import {shallow, ShallowWrapper} from "enzyme"
import {node} from "../../../test-util/SelectionUtil"

describe(AddItem.name, () => {

    it("displays the item form", () => {
        const tree = renderer.create(<AddItem name={"Chicken"} calories={"500"} addItem={jest.fn()} itemNameChanged={jest.fn()} itemCaloriesChanged={jest.fn()} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    describe("when", () => {
        let wrapper: ShallowWrapper, addItem: jest.Mock, itemNameChanged: jest.Mock, itemCaloriesChanged: jest.Mock

        beforeEach(() => {
            addItem = jest.fn()
            itemNameChanged = jest.fn()
            itemCaloriesChanged = jest.fn()
            wrapper = shallow(<AddItem name={"Chicken"} calories={"500"} addItem={addItem} itemNameChanged={itemNameChanged} itemCaloriesChanged={itemCaloriesChanged} />)
        })

        it("the add item button is pressed adds an item", () => {
            node(AddItemTestIds.AddButton, wrapper).props().onPress(null as any)
            expect(addItem).toHaveBeenCalled()
        })

        it("name text changes updates the name text", () => {
            node(AddItemTestIds.NameField, wrapper).props().onChangeText("Soup")
            expect(itemNameChanged).toHaveBeenCalledWith("Soup")
        })

        it("calorie text changes updates the calorie text", () => {
            node(AddItemTestIds.CalorieField, wrapper).props().onChangeText("200")
            expect(itemCaloriesChanged).toHaveBeenCalledWith("200")
        })

    })

})
