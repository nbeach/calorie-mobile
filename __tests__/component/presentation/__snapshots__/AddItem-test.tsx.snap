// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem displays the item form 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e4ebee",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="Item"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#AAAAAA",
        "borderStyle": "solid",
        "fontSize": 18,
        "width": "60%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    placeholder="Cals"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#AAAAAA",
        "borderStyle": "solid",
        "fontSize": 18,
        "width": "15%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1b5fff",
        "borderColor": "#EEEEEE",
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "justifyContent": "center",
        "opacity": 1,
        "width": "15%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
